server.go

```diff
@@ -364,6 +370,14 @@ func (s *Server) serveFile(w http.ResponseWriter, r *http.Request) {
 // Note that filedrop code is URL prefix-agnostic, so request URI doesn't
 // matters much.
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Run a cleanup in a goroutine only if not cleaning up at the moment
+	// and last cleanup was more than CleanupIntervalSeconds seconds ago.
+	if atomic.LoadInt64(&s.fileCleanerLastRun) != 0 {
+		if time.Now().Unix() - atomic.LoadInt64(&s.fileCleanerLastRun) > int64(s.Conf.CleanupIntervalSecs) {
```

Comment:
Note that this does not prevent multiple cleanup goroutines from being started. Two goroutines can load the same value (twice) and start cleanupFiles twice. You can try to use CompareAndSwap as a "lightweight mutex" here, if you really want.